--- origsrc/gmic-1.4.8.2/src/Makefile	2011-02-24 07:40:05.000000000 -0600
+++ src/gmic-1.4.8.2/src/Makefile	2011-03-03 23:30:26.626515100 -0600
@@ -148,9 +148,11 @@ else
 MAGICK_LDFLAGS = -lGraphicsMagick++
 endif
 
+ifneq ($(OPENEXR),no)
 # Flags to enable native support of EXR file format, using the OpenEXR library/
 EXR_CFLAGS = -Dcimg_use_openexr -I$(USR)/include/OpenEXR
 EXR_LDFLAGS = -lIlmImf -lHalf
+endif
 
 # Flags to enable the use of the FFTW3 library.
 FFTW_CFLAGS = -Dcimg_use_fftw3
