--- origsrc/gmic-1.5.0.0/src/Makefile	2011-08-11 17:07:25.170238300 -0500
+++ src/gmic-1.5.0.0/src/Makefile	2011-08-11 16:58:44.504457900 -0500
@@ -58,6 +58,9 @@ else
 PLUGINDIR = `gimptool-2.0 --gimpplugindir`/plug-ins
 USR = /usr
 endif
+ifeq ($(OS),Cygwin)
+EXE = .exe
+endif
 ifeq ($(OS),MINGW32_NT-5.1)
 EXE = .exe
 endif
@@ -244,6 +247,15 @@ ifeq ($(OS),Darwin)
 	$(MAKE) bashcompletion
 	$(MAKE) gimp
 else
+ifeq ($(OS),Cygwin)
+	@echo "**"
+	@echo "** Cygwin configuration"
+	@echo "**"
+	$(MAKE) linux
+	$(MAKE) bashcompletion
+	$(MAKE) gimp
+	$(MAKE) lib
+else
 	@echo "**"
 	@echo "** Windows configuration"
 	@echo "**"
@@ -252,6 +264,7 @@ else
 	$(MAKE) gimp
 endif
 endif
+endif
 
 gimp:
 	$(MAKE) "CFLAGS=$(STD_GIMP_CFLAGS) $(OPT_CFLAGS)" "LDFLAGS+=$(STD_GIMP_LDFLAGS) $(OPT_LDFLAGS)" "STRIP_EXE=1" gmic_gimp
@@ -289,8 +302,12 @@ gmic_lib.o: gmic.cpp gmic_def.h
 	$(CC) -o gmic_lib.o -c gmic.cpp -fPIC -Dgmic_minimal -Dgmic_float $(CFLAGS)
 gmic_lib: gmic_lib.o
 	ar rcs libgmic.a gmic_lib.o
+ifeq ($(OS),Cygwin)
+	$(CC) -shared -Wl,--out-implib,libgmic.dll.a -o cyggmic-1.dll gmic_lib.o $(LDFLAGS)
+else
 	$(CC) -shared -Wl,-soname,libgmic.so.1 -o libgmic.so gmic_lib.o $(LDFLAGS)
 	$(CC) -o gmic_use_lib gmic_use_lib.cpp -L. -lgmic $(FFTW_LDFLAGS)
+endif
 
 gmic_gimp.o: gmic.cpp gmic_def.h
 	$(CC) -o gmic_gimp.o -c gmic.cpp -Dgmic_gimp -Dgmic_minimal -Dgmic_float $(CFLAGS)
@@ -344,10 +361,16 @@ install:
 	cp -f gmic $(DESTDIR)$(USR)/bin/
 	mkdir -p $(DESTDIR)$(USR)/include/
 	cp -f gmic.h $(DESTDIR)$(USR)/include/
+ifeq ($(OS),Cygwin)
+	mkdir -p $(DESTDIR)$(USR)/lib
+	cp -f cyggmic-1.dll $(DESTDIR)$(USR)/bin/
+	cp -f libgmic.dll.a $(DESTDIR)$(USR)/lib/
+else
 ifneq ($(OS),Darwin)
 	mkdir -p $(DESTDIR)$(USR)/lib
 	cp -f libgmic.so $(DESTDIR)$(USR)/lib/libgmic.so.1
 endif
+endif
 	mkdir -p $(DESTDIR)$(USR)/share/man/man1/
 	mkdir -p $(DESTDIR)$(USR)/share/man/fr/man1/
 	cp -f ../man/gmic.1.gz $(DESTDIR)$(USR)/share/man/man1/gmic.1.gz
