--- origsrc/gmic-1.6.2.0/src/Makefile	2015-04-16 08:50:53.000000000 -0500
+++ src/gmic-1.6.2.0/src/Makefile	2015-04-21 21:45:59.951923400 -0500
@@ -411,7 +411,6 @@ gmic_gimp.o: gmic.cpp
 	$(CC) -o gmic_gimp.o -c gmic.cpp $(CFLAGS) -Dgmic_gimp
 gmic_gimp : gmic_gimp.o gmic_gimp.cpp
 	$(CC) -o gmic_gimp gmic_gimp.cpp gmic_gimp.o -Dgmic_gimp `gimptool-2.0$(EXE) --cflags` $(CFLAGS) `gimptool-2.0$(EXE) --libs` $(LIBS)
-	strip gmic_gimp$(EXE)
 
 gmic: gmic.cpp gmic.h CImg.h
 	$(CC) -o gmic gmic.cpp $(CFLAGS) -Dgmic_main $(LIBS)
