--- origsrc/gmic-1.6.0.3/src/Makefile	2014-12-19 04:51:53.000000000 -0600
+++ src/gmic-1.6.0.3/src/Makefile	2015-01-04 14:49:37.020058800 -0600
@@ -393,12 +393,16 @@ gmic_lib.o: gmic.cpp
 	$(CC) -o gmic_lib.o -c gmic.cpp -fPIC $(CFLAGS)
 gmic_lib: gmic_lib.o
 	ar rcs libgmic.a gmic_lib.o
+ifeq ($(OSTYPE),Cygwin)
+	$(CC) -shared -Wl,--out-implib,libgmic.dll.a -o cyggmic-1.dll gmic_lib.o $(LIBS)
+else
 ifneq ($(OS),Darwin)
 	$(CC) -shared -Wl,-soname,libgmic.so.1 -o libgmic.so gmic_lib.o $(LIBS)
 	$(CC) -o gmic_use_lib gmic_use_lib.cpp -L. -lgmic $(FFTW_LIBS)
 else
 	$(CC) -shared -o libgmic.so gmic_lib.o $(LIBS)
 endif
+endif
 
 gmic_gimp.o: gmic.cpp CImg.h
 	$(CC) -o gmic_gimp.o -c gmic.cpp $(CFLAGS) -Dgmic_gimp
@@ -438,6 +442,11 @@ install:
 	cp -f gmic $(DESTDIR)$(USR)/$(BIN)/
 	mkdir -p $(DESTDIR)$(USR)/$(INCLUDE)/
 	cp -f gmic.h $(DESTDIR)$(USR)/$(INCLUDE)/
+ifeq ($(OSTYPE),Cygwin)
+	mkdir -p $(DESTDIR)$(USR)/lib
+	cp -f cyggmic-1.dll $(DESTDIR)$(USR)/$(BIN)/
+	cp -f libgmic.dll.a $(DESTDIR)$(USR)/$(LIB)/
+else
 ifneq ($(OS),Darwin)
 	mkdir -p $(DESTDIR)$(USR)/share
 	mkdir -p $(DESTDIR)$(USR)/$(LIB)
@@ -445,6 +454,7 @@ ifneq ($(OS),Darwin)
 	ln -s libgmic.so.1.6.0 $(DESTDIR)$(USR)/$(LIB)/libgmic.so.1
 	ln -s libgmic.so.1 $(DESTDIR)$(USR)/$(LIB)/libgmic.so
 endif
+endif
 	mkdir -p $(DESTDIR)$(USR)/share/man/
 	mkdir -p $(DESTDIR)$(USR)/share/man/man1/
 	mkdir -p $(DESTDIR)$(USR)/share/man/fr/man1/
